File: examples/strongscript/2.ts
class A {
	constructor(public a) {}
	getA() { return this.a }
}

class B extends A {
	constructor(public a : !number) {super(a); }
	getA():!number { return this.a }
}

class C {
	constructor(public ref) {}
	main() {
		var ref:!A = new A(2)
		ref = new B(2)
		return ref.getA()
	}
}

File: examples/reticulated_python/2.py
@fields({'a' : Dyn})
class A:
	def __init__(self, a:Int):
		self.a = a
	def getA(self:A)->Dyn:
		return self.a

@fields({'a' : Int})
class B(A):
	a = 1
	def __init__(self:B, a:Int):
		self.a = a
	def getA(self:B)->Int:
		return self.a

@fields({'ref' : A})
class C:
	ref = A(1)
	def main(self):
		self.ref = A(2)
		self.ref = B(2)
		self.ref.getA()


C().main()
